core/settlement/engine.go

```diff
@@ -461,35 +462,29 @@ func (e *Engine) settleAll(assetDecimals uint32) ([]*types.Transfer, error) {
 	return aggregated, nil
 }
 
-// this doesn't need the mutex wrap because it's an internal call and the function that is being
-// called already locks the positions map.
-func (e *Engine) getCurrentPosition(party string, evt events.MarketPosition) *pos {
-	p, ok := e.pos[party]
+func (e *Engine) getOrCreateCurrentPosition(party string, size int64, price *num.Uint) (int64, *num.Uint) {
+	p, ok := e.settledPosition[party]
 	if !ok {
-		p = newPos(evt)
-		e.pos[party] = p
+		e.settledPosition[party] = size
+		return size, price
 	}
-	return p
-}
-
-func (e *Engine) AddPosition(party string, evt events.MarketPosition) {
-	_ = e.getCurrentPosition(party, evt)
+	return p, e.lastMarkPrice
 }
 
 func (e *Engine) HasPosition(party string) bool {
-	_, okPos := e.pos[party]
+	_, okPos := e.settledPosition[party]
```

Comment:
The behaviour here might have changed slightly. If a party has traded, but the position was zero, and we haven't marked them to market, wouldn't this call return `false`, whereas the party is actually holding a position, it's just that it hasn't been MTM'ed yet

Reply:
1. this is not a change of behaviour introduced in this PR
2. you're reading this right but it's not clear to me from the context of the call (releaseExcessMargin) if it expects only to get parties with settled position + unsettled trades (which is what it will get now) or parties with unsettled trades (with or without settled position).

as this change was introduced by @jeremyletang I'll leave it to  him to comment.